### PR TITLE
DDCNL-606: clean user cache after successfully submission of tamc

### PR DESCRIPTION
**Pull request by frikit**
add generate coverage report on run
upgrade coverage plugin from 1.5.1 -> 1.6.1

### CODE REVIEW

**Comment by frikit** [comment]
Please review @lewey @pascal22p @danewatts

**Comment by frikit** [comment]
also some minor refactor

File: app/controllers/TransferController.scala [hunk 1]
```diff
@@ -240,6 +240,7 @@ class TransferController @Inject()(
     implicit request =>
       registrationService.getFinishedData(request.nino) map {
         case NotificationRecord(email) =>
+          cachingService.remove()
```
Comment (danewatts) [inline app/controllers/TransferController.scala]: Can you add some unit tests with a verify step to ensure the cache is called only in this scenario and is not called in the others (Mockito verify checks)
We should also consider if we want any logging when removing from the cache fails (recover)

**Comment by frikit** [comment]
fix please re-review @danewatts

File: test/controllers/TransferControllerTest.scala [hunk 2]
```diff
@@ -390,12 +389,38 @@ class TransferControllerTest extends ControllerBaseSpec {
 
   "finished" should {
     "return success" when {
-      "A notification record is returned" in {
+      "make sure method remove never was called before" in {
+        verify(mockCachingService, times(0)).remove()(any(), any())
```
Comment (danewatts) [inline test/controllers/TransferControllerTest.scala]: You arent calling anything for the verify to work here?

File: test/controllers/TransferControllerTest.scala [hunk 3]
```diff
@@ -390,12 +389,38 @@ class TransferControllerTest extends ControllerBaseSpec {
...
+      "A notification record is returned and cache is called" in {
         when(mockTransferService.getFinishedData(any())(any(), any()))
           .thenReturn(RelationshipRecordData.notificationRecord)
+
         val result = controller().finished()(request)
         status(result) shouldBe OK
       }
+
+      "make sure method remove was called only once" in {
+        verify(mockCachingService, times(1)).remove()(any(), any())
```
Comment (danewatts) [inline test/controllers/TransferControllerTest.scala]: Unit tests need to be independent. This test specifically requires previous test orders.

File: test/controllers/TransferControllerTest.scala (hunk 2 above)
Comment (frikit) [reply test/controllers/TransferControllerTest.scala]: I split one test in 3 test(3 steps just to be readable)

verify method is checking as method remove was called 0 times

and next step I call controller().finished() which actual contain inside .remove() method and then I check as this method was called exactly once

**Comment by frikit** [comment]
As we discuss, reset mock's on each test, also make 2 test not 6

**Comment by danewatts** [comment]
+1. Going forward can you please prefix commit message with branch name. 
EG: DDCNL-606: Add test for blah blah

This will help link it to JIRA.

Please **SQUASH** this before merging and amend to appropriate commit

**Comment by frikit** [comment]
@danewatts first commit have this prefix, after squash will be fine

File: test/services/TimeServiceTest.scala [hunk 4]
```diff
@@ -86,11 +86,48 @@ class TimeServiceTest extends ControllerBaseSpec {
...
+
+    "parse date with default format of yyyyMMdd" in {
+      val year = 2017
+      val month = 10
+      val day = 22
+      val expected = new LocalDate(year, month, day)
+
+      val date = "" + year + month + day
+
+      service.parseDateWithFormat(date) shouldBe expected
```
[the remark this answers is not in the document]
Comment (Tateni93) [reply test/services/TimeServiceTest.scala]: https://github.com/hmrc/tcs-frontend/blob/master/test/testutils/TestUtils.scala#L30
https://github.com/hmrc/tcs-frontend/blob/b6f938b3e1c50df880ef2274fcd85216b2293600/test/controllers/workandhours/partials/stillhaveajobanswer/PartnerSpec.scala#L67

**Pull request merged**